test/TokenVault.test.ts

```diff
@@ -25,52 +21,52 @@ describe('Vault', () => {
         const IERC20Artifact = await artifacts.readArtifact('IERC20');
         token = await deployMockContract(wallet1 as Signer, IERC20Artifact.abi);
 
-        const VaultFactory = await ethers.getContractFactory('Vault', wallet1);
-        vault = await VaultFactory.deploy(wallet1.address);
+        const TokenVaultFactory = await ethers.getContractFactory('TokenVault', wallet1);
+        tokenVault = await TokenVaultFactory.deploy(wallet1.address);
     });
 
     describe('constructor', () => {
         it('should set the owner', async () => {
-            expect(await vault.owner()).to.equal(wallet1.address)
+            expect(await tokenVault.owner()).to.equal(wallet1.address)
         })
     })
 
     describe('setApproval()', () => {
         it('should allow the owner to approve accounts', async () => {
-            await vault.setApproved(wallet2.address, true)
-            expect(await vault.approved(wallet2.address)).to.equal(true)
+            await tokenVault.setApproved(wallet2.address, true)
+            expect(await tokenVault.approved(wallet2.address)).to.equal(true)
         })
     })
 
     describe('increaseERC20Allowance()', () => {
         it('should allow owners to increase approval amount', async () => {
-            await vault.setApproved(wallet2.address, true)
-            await token.mock.allowance.withArgs(vault.address, wallet2.address).returns('0')
+            await tokenVault.setApproved(wallet2.address, true)
+            await token.mock.allowance.withArgs(tokenVault.address, wallet2.address).returns('0')
             await token.mock.approve.withArgs(wallet2.address, '1111').returns(true)
-            await vault.increaseERC20Allowance(token.address, wallet2.address, '1111')
+            await tokenVault.increaseERC20Allowance(token.address, wallet2.address, '1111')
```

Comment:
You know what is interesting about this test- is that it doesn't assert the state change.

The mock doesn't have an expectation, so if the mock isn't called this test will still pass.

In this case, the easiest thing would be to instantiate a real token and check to see that the allowance was changed properly

Reply:
Done: https://github.com/pooltogether/v4-core/pull/272/commits/b9457454673816223dca65caaf45b6fa23e2b5a0